README.md

```diff
@@ -112,6 +112,88 @@ If the `current-context` is set to the name of one of the clusters then the
 `--context` argument may be omitted, and the cluster named by `current-context`
 will be used.
 
+## Kubernetes (example)
+Kuberos can be ran inside a cluster as long as it can still communicate with your OIDC provider from inside the pod and your OIDC provider is set to redirect to your Kuberos endpoint (NodePort, LoadBalancer, etc).
+
+The configuration below is meant to serve as a template and **not** something that is plug-and-play. You will need to adjust your DNS / nameserver helpers, Dex information, and optionally how you ingress your traffic.
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: kuberos
+  name: kuberos
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kuberos
+    spec:
+      hostAliases:                  #
+      - ip: "192.168.1.1"           # OPTIONAL TO HELP ROUTE TRAFFIC TO DEX / OIDC
+        hostnames:                  #
+        - "dex.oidc.example.com"    # OPTIONAL TO HELP ROUTE TRAFFIC TO DEX / OIDC
+      containers:
+      - image: negz/kuberos:latest
+        name: kuberos
+        command: ["/kuberos", "https://dex.oidc.example.com", "example-app", "/cfg/secret", "/cfg/template"]
+        ports:
+        - name: http
+          containerPort: 10003      # DEFAULT FOR KUBEROS 10003, MATCHES NODEPORT SERVICE
+        volumeMounts:
+        - name: config
+          mountPath: /cfg
+      dnsConfig:
+        nameservers:
+          - 8.8.8.8                 # USING GOOGLE DNS FOR LOOKUPS
+      volumes:
+      - name: config
+        configMap:
+          name: kuberos
+          items:
+          - key: template
+            path: template
+          - key: secret
+            path: secret
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: kuberos
+data:
+  template: |
+    apiVersion: v1
+    kind: Config
+    current-context: staging
+    clusters:
+    - name: production
+      cluster:
+        certificate-authority-data: REDACTED
+        server: https://prod.example.org
+    - name: staging
+      cluster:
+        certificate-authority-data: REDACTED
+        server: https://staging.example.org
+  secret: ZXhhbXBsZS1hcHAtc2VjcmV0
+---
+apiVersion: v1
+kind: Service
```

Comment:
Perhaps we should leave the service as an exercise for the reader? I feel like these things vary too much from company to company (i.e. some might use an ingress controller, others `type: NodePort`, others `type: LoadBalancer`) for an example to provide much illustration.

Reply:
Agreed. I haven't done a deep dive on your code to see if there was any issues with a cloud LB or doing something like SSL termination on an ingress controller. If it is a simple client / server model, and you see no issues with it, I think it makes sense to leave it up to the operator and just notate that.